frontend/src/pages/TaskMapPage.tsx

```diff
@@ -153,8 +153,13 @@ export const TaskMapPage = () => {
     dispatch(roadmapsActions.getRoadmaps());
   };
 
-  const addSynergyRelations = (from: number, to: number[]) =>
-    dispatch(roadmapsActions.addSynergyRelations({ from, to })).unwrap();
+  const addSynergyRelations = async (from: number, to: number[]) => {
+    const ok = await dispatch(
+      roadmapsActions.addSynergyRelations({ from, to }),
+    ).unwrap();
+    if (ok) dispatch(roadmapsActions.getRoadmaps());
+    return ok;
+  };
 
   const onDragMoveOutside = async (draggedTaskId: number) => {
     const newList = taskRelations
```

Comment:
The original issue seems to be fixed 👍 (_successfully drag a task to a group, navigate elsewhere and back to taskmap_)
But I found out that it also happens when dragging task out of a group. But simply adding `dispatch(roadmapsActions.getRoadmaps());` to the end of the onDragMoveOutside() function should fix that

Reply:
`onConnect()` and `addSynergyRelations()` are the only functions that dispatch api actions to the server, and both refresh the roadmaps at the end. The issue was that the api checked only for 200, when the server returned 204 (no content).

I fixed this for all the api calls that just check the http status.

Reply:
Good find 👍